tests/all/custom_signal_handler.rs

```diff
@@ -1,6 +1,9 @@
 #[cfg(any(
     target_os = "linux",
-    all(target_os = "macos", feature = "posix-signals-on-macos")
+    all(
+        any(target_os = "macos", target_os = "ios"),
+        feature = "posix-signals-on-macos"
+    )
 ))]
 #[cfg(not(miri))]
```

Comment:
As a heads up I've posted https://github.com/bytecodealliance/wasmtime/pull/7509 which will conflict with this, although what you're doing here of "where macos is needed also allow ios" would also apply to that PR.